met-api/src/met_api/resources/survey.py

```diff
@@ -58,11 +58,20 @@ class Survey(Resource):
     @staticmethod
     # @TRACER.trace()
     @cross_origin(origins=allowedorigins())
+    @auth.requireOptional
     def get(survey_id):
         """Fetch a single survey matching the provided id."""
         try:
-            survey_record = SurveyService().get(survey_id)
-            return ActionResult.success(survey_id, survey_record)
+            user_id = TokenInfo.get_id()
+            if user_id:
```

Comment:
Trying to think of a cleaner way than having a flag in the method that decides which method is called. Maybe we can have that logic in the service at least. The resource shouldn't care about that particular business logic

Reply:
We can call one service method and it will be a small concise method def get_survey(survey_id, user_id) {

if(user_id) return get(survey_id)

return get_open(survey_id).

This way the resource doesn't have to manage this logic

Reply:
I'll make this change but it also needs to be changed in the future, I mentioned this before but the right approach would be to have separate API endpoints like "public" and "internal" and not a common endpoint that will do both!